immutable_tree.go

```diff
@@ -146,19 +146,26 @@ func (t *ImmutableTree) Export() *Exporter {
 // otherwise. The returned value must not be modified, since it may point to data stored within
 // IAVL.
 // TODO: Understand what is this index? Index on its own isn't well defined
-// index across all leaves?
+// index = index of the leaf node
```

Comment:
But index amongst what? Like between the `m` branches in a BTree?

Reply:
"The index is the index in the list of leaf nodes sorted lexicographically by key. The leftmost leaf has index 0. It's neighbor has index 1 and so on." I noticed this in the docs found [here](https://github.com/cosmos/iavl/blob/master/docs/tree/immutable_tree.md#get)

Reply:
I'll add this in a comment in the codebase so that is immediately clear what index represents.

Reply:
Sweet, tysm